ToDo/src/components/app/app.js

```diff
@@ -1,31 +1,140 @@
-import React from 'react'
-import TodoList from '../todo-list'
-import Header from '../header'
-import SearchField from '../search-field'
-import ItemStatusFilter from '../item-status-filter'
+import React, {Component} from 'react'
+import TodoList from 'components/todo-list'
+import Header from 'components/header'
+import SearchField from 'components/search-field'
+import ItemStatusFilter from 'components/item-status-filter'
+import FormAddItem from 'components/form-add-item'
 import './app.css'
 
-const App = () => {
-    const todoData = [
-        {label:'Cooking food', important:false, id:1},
-        {label:'Drink tee', important:false, id:2},
-        {label:'Learn React', important:true, id:3},
-    ];
-    const style = {
-        margin: '0 auto',
-        maxWidth: '600px'
+export default class App extends Component {
+    maxId = 100;
+    state = {
+        todoData : [
+            this.createTodoItem('Cooking food'),
+            this.createTodoItem('Drink tee'),
+            this.createTodoItem('Learn React'),
+        ],
+        term: '',
+        filter: 'all'
     };
+    createTodoItem(label) {
+        return {
+            label,
+            important: false,
+            done: false,
+            id: this.maxId++,
+        }
+    };
+    deleteItem = (id) => {
+        this.setState(({ todoData }) => {
+            const index = todoData.findIndex((el) => el.id === id);
+            const newArr = [
+                ...todoData.slice(0, index),
+                ...todoData.slice(index+1)
```

Comment:
По форматированию:
```suggestion
                ...todoData.slice(index + 1)
``` Более читабельно и в редакторе не сливается)

Reply:
done

Reply:
done